src/filesystem.ts

```diff
@@ -1,82 +1,72 @@
 import { CID } from "multiformats/cid"
 
-import FileSystem from "./fs/index.js"
-
-import * as cidLog from "./common/cid-log.js"
-import * as common from "./common/index.js"
-import * as crypto from "./crypto/index.js"
-import * as debug from "./common/debug.js"
-import * as dataRoot from "./data-root.js"
-import * as did from "./did/index.js"
-import * as pathing from "./path.js"
-import * as protocol from "./fs/protocol/index.js"
-import * as storage from "./storage/index.js"
-import * as token from "./ucan/token.js"
-import * as ucan from "./ucan/internal.js"
-import * as versions from "./fs/versions.js"
-
-import { Branch } from "./path.js"
-import { Maybe, authenticatedUsername, decodeCID } from "./common/index.js"
-import { Permissions } from "./ucan/permissions.js"
-import { setup } from "./setup/internal.js"
+import * as Depot from "./components/depot/implementation.js"
+import * as Path from "./path/index.js"
+import * as Reference from "./components/reference/implementation.js"
+
+import * as Protocol from "./fs/protocol/index.js"
+import * as Versions from "./fs/versions.js"
+
+import FileSystem, { Dependents } from "./fs/filesystem.js"
+
+import { Branch } from "./path/index.js"
+import { Configuration } from "./configuration.js"
+import { Maybe, decodeCID, EMPTY_CID } from "./common/index.js"
 
 
 /**
  * Load a user's file system.
- *
- * @param permissions The permissions from initialise.
- *                    Pass `null` if working without permissions
- * @param username Optional, username of the user to load the file system from.
- *                 Will try to load the file system of the authenticated user
- *                 by default. Throws an error if there's no authenticated user.
- * @param rootKey Optional, AES key to be the root key of a new filesystem.
- *                Will be used if a filesystem hasn't been created yet.
  */
-export async function loadFileSystem(
-  permissions: Maybe<Permissions>,
-  username?: string,
-  rootKey?: string
-): Promise<FileSystem> {
+export async function loadFileSystem({ config, dependents, rootKey, username }: {
+  config: Configuration
+  dependents: Dependents
```

Comment:
I'm playing around with this a little right now, and that seemed off.

Aren't these `dependencies`, not `Dependents`?
I.e. the `FileSystem` depends on `crypto`, `storage`, `reference`, etc. not the other way around :thinking:

Reply:
Hmm true, I'll change it 👍 
By the way, in case you hadn't seen this, there's a version of this function available on the `Program` that doesn't require to pass in components.